### ISSUE TITLE
refresh token rotation doesn't update client session 

### ISSUE THREAD

**Issue by Cantinaband** [nextauthjs/next-auth#6710]
### Question 💬

Hi, 

I'm following this guide to refresh the token. 

https://authjs.dev/guides/basics/refresh-token-rotation

The refresh function works as expected and get a fresh `access_token` & new `refresh_token`. The problem is that the client session isn't updated. So every time the JWT callback function is called, the token argument is the old, expired token, and the refreshToken function will be called every time. I would expect it to be called only once, and then it should be fine until the token expires again. But that's not the case. 

After i called `/api/auth/session` on the client the session is up to date... 
### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by shammlo** [comment]
after the refreshing tokens, does it update the token? or somehow it still uses the old one? sorry if its not related to your question or anything.

**Comment by FernandVEYRIER** [comment]
Facing the same issues here. It happens on page refresh, where the `jwt` callback is called once with the new token and the old token.

**Comment by fabius** [comment]
Facing the same issue using SvelteKit.

The setup as per configuration, specifically `session = event.locals.getSession()` inside your `*.server.ts` files seems to misbehaving. Just like mentioned before, manually calling the endpoint works as expected:

```ts
// src/routes/+layout.server.ts
import type { LayoutServerLoad } from './$types'

export const load: LayoutServerLoad = async (event) => {
  return {
    // session: await event.locals.getSession(),
    // faulty behaviour ^^^^^^^^^^^^^^^^^^^^^^^

    // this works fine
    session: (await event.fetch('/auth/session')).json(),
  }
}

```

**Comment by half2me** [comment]
@fabius I'm facing the same issue. I also just manually call the `auth/session` endpoint and that works well.

**Comment by luismiguelprs** [comment]
same issue here...

**Comment by javsco** [comment]
getting the same error using the credentials provider.

**Comment by balazsorban44** [comment]
Please, read https://github.com/nextauthjs/next-auth/issues/6710#issuecomment-1427951217

**Comment by userdansilva** [comment]
@javsco me too, do you find a solution for it?

**Comment by louisthomaspro** [comment]
Same here, please fix it!

**Comment by Cantinaband** [comment]
After we added the SessionProvider to the App the Token Rotation works just fine.

**Comment by balazsorban44** [comment]
Closing as no reproduction was added. If you have one, please open a new issue.
